django/core/management/templates.py

```diff
@@ -74,6 +74,8 @@ def handle(self, app_or_project, name, target=None, **options):
             except OSError as e:
                 raise CommandError(e)
         else:
+            if app_or_project == 'app':
+                self.validate_name(os.path.basename(target))
```

Comment:
I'm not sure about using `validate_name()` here, it is a bit confusing, moreover messages should be different e.g.
- `'{target}' is not a valid {app} directory. Please make sure the directory is a valid identifier.`
- `'{target}' conflicts with the name of an existing Python module and cannot be used as {an} {app} directory. Please try another directory.` (this is untested).

Maybe we can add some extra flag to distinguish messages.